cmd/koinos-jsonrpc/main.go

```diff
@@ -232,5 +272,6 @@ func main() {
 	ch := make(chan os.Signal, 1)
 	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
 	<-ch
+	close(jobChan)
```

Comment:
I think this could lead to unclean/delayed shutdown.

The job go routines will see the `!ok` on the channel (but only after the channel has drained). In the meantime, the `httpHandler` will still be trying to send to the channel.

I think the better way of handling it would be with a context.

```
ctx, ctxCancel := context.WithCancel(context.Background())
defer ctxCancel()
```

Then, when you catch the signal and exit `main`, the context will be cancelled. So long as all go routines also wait on consuming from `ctx.Done()`, when the context is cancelled they will consume from that channel which will lead to an exit.